ui/lib/kubernetes/addon/routes/fetch-config.js

```diff
@@ -0,0 +1,25 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class KubernetesFetchConfigRoute extends Route {
+  @service store;
+  @service secretMountPath;
+
+  configModel = null;
+
+  async beforeModel() {
+    const backend = this.secretMountPath.get();
+    // check the store for record first
+    this.configModel = this.store.peekRecord('kubernetes/config', backend);
```

Comment:
No as soon as the config is created and saved it will be in the store. If this check does not return a record it will then go to the server via `store.queryRecord` to look for it so this just prevents going to the server after we already have the config loaded.

Reply:
I missed below where you `queryRecord`!